### PR TITLE
Ike v1 logging tx transforms 5455 v3

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/5455

Describe changes:
- ikev1 : log fields from transaction instead of fields from state

Replaces #7668 with using u64 instead of usize

### CODE REVIEW

File: rust/src/ike/ikev1.rs [hunk 1]
```diff
@@ -40,27 +40,27 @@ pub struct IkeV1Header {
 pub struct Ikev1ParticipantData {
     pub key_exchange: String,
     pub nonce: String,
-    pub vendor_ids: HashSet<String>,
-    /// nested Vec, outer Vec per Proposal/Transform, inner Vec has the list of attributes.
-    pub transforms: Vec<Vec<SaAttribute>>,
+    pub nb_transforms: u64,
```
Comment (victorjulien) [inline rust/src/ike/ikev1.rs:43]: so what happens if we actually get some really high number in here? the u64 suggests we'd handle it but given the loops of the actual transforms vec I'm not so sure

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 8457

File: rust/src/ike/ikev1.rs (hunk 1 above)
Comment (catenacyber) [reply rust/src/ike/ikev1.rs:43]: To reach it, someone needs to send at least U64_MAX bytes of data...

The worst that would happen then, is that we would stop logging the transform chosen by the server (which is controlled by the attacker in this scenario)

**Comment by victorjulien** [comment]
Merged in #7713, thanks!

**Pull request closed without merging**